flycheck.el

```diff
@@ -5440,9 +5441,10 @@ ERR is in BUFFER-FILES, replace it with the return value of the
 function `buffer-file-name'."
   (flycheck-error-with-buffer err
     (-when-let (filename (flycheck-error-filename err))
-      (when (seq-some (apply-partially #'flycheck-same-files-p
-                                       (expand-file-name filename cwd))
-                      buffer-files)
+      (when (or (seq-some (apply-partially #'flycheck-same-files-p
+                                           (expand-file-name filename cwd))
+                          buffer-files)
+                (not buffer-files))
```

Comment:
I added this or condition to make the tests pass. Since I wasn't building with any temporary files, `buffer-files` was always coming through `nil`, so `seq-some` would always be false. This resulted in no errors making it back to the assertion. It's weird that this worked fine in real world usage.